iroha/src/sumeragi.rs

```diff
@@ -1811,7 +1811,8 @@ pub mod config {
         /// # Errors
         /// Fails if there is no file or if file is not valid json
         pub fn from_path<P: AsRef<Path> + Debug>(path: P) -> Result<TrustedPeers> {
-            let file = File::open(path).wrap_err("Failed to open a file")?;
+            let error = format!("Failed to open trusted peers file {:?}", &path);
```

Comment:
Format will be always executed here. You can pass closure to wrap error. There is method called wrap_err_with

Reply:
But we are losing ownership on the path when we are trying to open file. I've described it in PR message.

Reply:
You can add there a copy and pass reference

Reply:
https://doc.rust-lang.org/std/convert/trait.AsRef.html#impl-AsRef%3CPath%3E-2

Reply:
Maybe you mean `unwrap_or_else()`, not `wrap_err_with()`? I can't find the later anywhere :)

Reply:
Here it is:
https://github.com/hyperledger/iroha/blob/iroha2-dev/iroha_error/src/wrap_err.rs#L76-L79

Reply:
And another question: wrap_err is printing error to stdout, but how do I print it from closure? Not just `println!()` :)

Reply:
It doesn't print error to stdout. Error is printed by rust wrapper from here:
https://github.com/hyperledger/iroha/blob/iroha2-dev/iroha_cli/src/main.rs#L14

As you can see `main` function returns `Result<(), Reporter>`. `Reporter` type actually prints errors to stdout. This wrapping thing is only about adding context to errors.

Reply:
Btw, also there is no need to pass reference to format. Display and Debug won't take ownership (unlike `dbg!` which actually returns all values it takes from args)

Reply:
Yep, I've used the `wrap_err_with()`, interesting way to fail :)

Reply:
Pushed the new version.